core/trino-main/src/main/java/io/trino/security/AccessControl.java

```diff
@@ -249,6 +249,13 @@
      */
     void checkCanDropColumn(SecurityContext context, QualifiedObjectName tableName);
 
+    /**
+     * Check if identity is allowed to alter columns to the specified table.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    void checkCanAlterColumn(SecurityContext context, QualifiedObjectName tableName);
```

Comment:
@martint @kokosing Do you think if we should check the permission in `ALTER COLUMN` level? Or it should be `SET DATA TYPE` (= `checkCanSetColumnType`) level?

Reply:
I wonder if `checkCanAlterTable` is not enough here. However this can be modeled by the implementation of the access control easily. So I think `checkCanAlterColumn` is good.

Reply:
According to the SQL spec:

> 1) The enabled authorization identifiers shall include the `<authorization identifier>` that owns the schema identified by the `<schema name>` of the table identified by `<table name>`.